### PR TITLE
Reader refactoring

### PR DESCRIPTION
**Pull request by trishorts**

### CODE REVIEW

File: mzLib/Readers/IDataReader.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿using Chemistry;
+using MzLibUtil;
+using MassSpectrometry; 
+namespace Readers
+{
+    // only loads file.
+    public interface IDataReader
+    {
+        public string FilePath { get; }
+        public void LoadAllStaticData(FilteringParams filteringParams = null, int maxThreads = 1);
```
Comment (trishorts) [inline mzLib/Readers/IDataReader.cs:10]: shouldn't be here

**Comment by acesnik** [comment]
image

File: mzLib/Test/TestReaders/TestMgf.cs [hunk 2]
```diff
@@ -55,19 +75,32 @@ public static void TestLoadMgf()
...
+            //There are two scans with scan #3. This is not allowed and an MzLibException is thrown
+            Assert.Throws<MzLibException>(() => dataReader.InitiateDynamicConnection());
+        }
+
+        [Test]
+        public static void TestLoadMgfTabSeparated()
+        {
+            string path = Path.Combine(TestContext.CurrentContext.TestDirectory, "DataFiles", "Tab_separated_peak_list.mgf");
+            var dataReader = ReaderCreator.CreateReader(path); 
+            dataReader.LoadAllStaticData();
```
Comment (acesnik) [inline mzLib/Test/TestReaders/TestMgf.cs:100]: It's apparently having trouble finding this file. Looks like it's named "Tab_separated_peak_list - Copy.mgf" in the file tree, which is probably the issue.

**Comment by acesnik** [comment]
I like the refactoring idea! Nice work simplifying the MS data file loading syntax.

**Pull request closed without merging**